history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/SimpleHistoryServiceImpl.java

```diff
@@ -10,48 +10,68 @@
 import pro.taskana.common.api.TaskanaRole;
 import pro.taskana.common.api.exceptions.InvalidArgumentException;
 import pro.taskana.common.api.exceptions.NotAuthorizedException;
-import pro.taskana.simplehistory.impl.mappings.HistoryEventMapper;
-import pro.taskana.simplehistory.impl.mappings.HistoryQueryMapper;
-import pro.taskana.simplehistory.query.HistoryQuery;
+import pro.taskana.simplehistory.impl.task.TaskHistoryEventMapper;
+import pro.taskana.simplehistory.impl.task.TaskHistoryQuery;
+import pro.taskana.simplehistory.impl.workbasket.WorkbasketHistoryEventMapper;
+import pro.taskana.simplehistory.impl.workbasket.WorkbasketHistoryQuery;
 import pro.taskana.spi.history.api.TaskanaHistory;
-import pro.taskana.spi.history.api.events.TaskanaHistoryEvent;
+import pro.taskana.spi.history.api.events.task.TaskHistoryEvent;
+import pro.taskana.spi.history.api.events.workbasket.WorkbasketHistoryEvent;
 import pro.taskana.spi.history.api.exceptions.TaskanaHistoryEventNotFoundException;
 
 /** This is the implementation of TaskanaHistory. */
 public class SimpleHistoryServiceImpl implements TaskanaHistory {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(SimpleHistoryServiceImpl.class);
   private TaskanaHistoryEngineImpl taskanaHistoryEngine;
-  private HistoryEventMapper historyEventMapper;
-  private HistoryQueryMapper historyQueryMapper;
+  private TaskHistoryEventMapper taskHistoryEventMapper;
+  private WorkbasketHistoryEventMapper workbasketHistoryEventMapper;
 
   @Override
   public void initialize(TaskanaEngineConfiguration taskanaEngineConfiguration) {
 
     this.taskanaHistoryEngine = getTaskanaEngine(taskanaEngineConfiguration);
+
     if (LOGGER.isDebugEnabled()) {
       LOGGER.debug(
           "Simple history service implementation initialized with schemaName: {} ",
           taskanaEngineConfiguration.getSchemaName());
     }
 
-    this.historyEventMapper =
-        this.taskanaHistoryEngine.getSqlSession().getMapper(HistoryEventMapper.class);
-    this.historyQueryMapper =
-        this.taskanaHistoryEngine.getSqlSession().getMapper(HistoryQueryMapper.class);
+    this.taskHistoryEventMapper =
+        this.taskanaHistoryEngine.getSqlSession().getMapper(TaskHistoryEventMapper.class);
+    this.workbasketHistoryEventMapper =
+        this.taskanaHistoryEngine.getSqlSession().getMapper(WorkbasketHistoryEventMapper.class);
   }
 
   @Override
-  public void create(TaskanaHistoryEvent event) {
+  public void create(TaskHistoryEvent event) {
     try {
       taskanaHistoryEngine.openConnection();
       if (event.getCreated() == null) {
```

Comment:
I think we could remove this check? The created timestamp gets set in every constructor